decode_test.go

```diff
@@ -4005,3 +4005,37 @@ func TestIssue384(t *testing.T) {
 		}
 	}
 }
+
+type EsDocument interface {
+	IndexName() string
+}
+
+type Property struct {
+	Name string `json:"name"`
+}
+
+func (p Property) IndexName() string {
+	return "properties"
+}
+
+func TestIssue405(t *testing.T) {
+	b := []byte(`{"name":"Test Property to remove"}`)
+	p := Property{}
+
+	func(body []byte, doc EsDocument) {
+		if err := json.Unmarshal(body, &doc); err != nil {
+			t.Errorf("unexpected error: %v", err)
+		}
+	}(b, &p)
+}
+
+func TestIssue406(t *testing.T) {
```

Comment:
Inside the function in Issue405, please create branches  Unmarshal and stream decoding.

```go
func TestIssue405(t *testing.T) {
  t.Run("unmarshal", func(t *testing.T) {

  }
  t.Run("stream", func(t *testing.T) {

  }
}
```

Reply:
Also please add a test case for encoding/json and make sure it's ok there

Reply:
TestIssue406 is still there, please remove this